test/player_unit.js

```diff
@@ -422,7 +422,6 @@ describe('Player', () => {
             dispatchEventSpy.calls.reset();
             player.configure({streaming: {maxDisabledTime}});
             player.setMaxHardwareResolution(123, 456);
-            onErrorCallback(httpError);
```

Comment:
Just to make sure I understand, was this line moved to the `it()` level since there's now a nested `describe()` scenario that also needs to call `onErrorCallback(httpError)` at the appropriate time in the jasmine test life cycle?

Reply:
Yes.  The original author of these tests put onErrorCallback() in beforeEach().  But since that is the thing which triggers the code under test, it really belongs in the test itself.  The rest can still be considered test setup.

This only became an issue when I wanted to nest another level with different setup.  All setup needs to happen before the code under test is triggered.